README.md

```diff
@@ -53,12 +58,43 @@ To add a new project type, the developer must do the following:
 3. Some function that will be run each time one of these types is
    encountered (that is, a required field) must also set the fields
    `github_name`, `github_email`, and `github_repo` if they are not in
-   the input JSON.
+   the input JSON.  (You may wish to override even if they are.)
    * `github_name` is the name of the author, and an arbitrary string.
    * `github_email` is the email address of the author, and is also
      arbitrary. 
-   * `github_repo` is of the form <github_org>/<github_repo>,
+   * `github_repo` is of the form `<github_org>/<github_repo>`,
      e.g. `lsst-sqre/uservice-mymicroservice`.
-4. Write unit tests for your field substitution in `tests`.
-	 
-See `uservice_ccutter/plugins/substitute.py` for more.
+4. `github_description` and `github_homepage` will populate those fields
+     on the GitHub repository; if `github_description` does not exist
+     but `description` does, `description` will be used instead.
+5. Write unit tests for your field substitution in `tests`.
+6. Add a `finalize_` function for work that needs to be done after
+   the push to GitHub, if any.
+
+See `uservice_ccutter/plugins/substitute.py` for more information on
+field substitution.
+
+#### Function Naming Conventions
+
+In your `<typename>.py` file, field names are mapped verbatim to
+function names, except that dashes in field names are replaced with
+underscores in function names, due to Python naming requirements.
+
+Functions ending with a single underscore are reserved for use by the
+plugin machinery, e.g. `finalize_`.
+
+#### The `finalize_` function
+
+A project type may need to perform actions after its GitHub repository
+has been created.  It does this in a function called `finalize_`.  If a
+project type does not have any post-commit actions, it may omit the
+function.
+
+If your `finalize_` function itself needs to call other functions, those
+functions should be named with a single leading underscore, to protect
+them from interpretation as field names.
```

Comment:
Finalize isn't called, because the exception from create_project isn't inside a try() so Flask will catch it and return an error to the user via @app.errorhandler(BackendError).